[WARNING] Oh My Zsh core files may be missing during cloud-init setup

## Issue Description
During CloudShell VM cloud-init execution, Oh My Zsh installation generates a warning about potentially missing core files despite the directory existing.

## Log Evidence
**Line 5884:**
```
[2025-08-22 22:46:40] [WARN] [dotfiles install.sh] Oh My Zsh directory exists but core files may be missing
```

## Impact
- **Severity:** Low
- **User Experience:** Oh My Zsh may not function properly
- **Functionality:** Terminal features and themes may be incomplete

## Root Cause Analysis
- Oh My Zsh directory exists from previous installation
- Core files validation failing during dotfiles setup
- Possible incomplete or corrupted previous installation

## Current Workaround
- System continues to function
- Oh My Zsh basic functionality may still work
- User can manually reinstall if needed

## Recommended Solutions
1. **Force Reinstall:** Remove existing directory and perform clean install
2. **File Integrity Check:** Verify and restore missing core files
3. **Installation Validation:** Add comprehensive post-install verification

## Additional Context
- Detected during automated cloud-init analysis on August 23, 2025
- Part of dotfiles installation process
- Related to terminal environment setup

## Priority
**Low** - Cosmetic issue affecting terminal experience but not core functionality

## Status Update: Still Occurring ⚠️

**Deployment**: August 24, 2025 - CLOUDSHELL VM  
**Status**: Issue persists in current deployment

### Log Evidence:
```
[2025-08-24 19:37:44] [ERROR] [dotfiles install.sh] Oh My Zsh directory exists but core files are missing
```

### Current Impact:
- Oh My Zsh installation validation failing
- Terminal themes and features may be incomplete
- User experience affected but system functional

### Analysis:
The dotfiles installation process detects an existing Oh My Zsh directory but core file validation is failing, suggesting either:
- Incomplete previous installation
- Missing critical Oh My Zsh files
- Validation logic too strict

This remains an active issue requiring attention for proper terminal environment setup.

--- a/cloud-init/CLOUDSHELL.conf
+++ b/cloud-init/CLOUDSHELL.conf
@@ -601,6 +601,14 @@ runcmd:
     echo "export ENABLE_BACKGROUND_TASKS=1" >> /root/.bashrc
     echo "export ENABLE_BACKGROUND_TASKS=1" >> /root/.zshrc
   - |
+    # Clean up any existing Oh My Zsh installation to prevent dotfiles conflicts
+    # This addresses GitHub issue #319 where existing .oh-my-zsh directory causes
+    # dotfiles installation to fail with "core files missing" error
+    if [ -d "/root/.oh-my-zsh" ]; then
+      echo "Removing existing Oh My Zsh directory to ensure clean dotfiles installation..."
+      rm -rf /root/.oh-my-zsh
+    fi
+  - |
     if git clone https://github.com/40docs/dotfiles.git /root/dotfiles; then
       cd /root/dotfiles
       export NVM_DIR="/usr/local/nvm"
@@ -1062,9 +1070,13 @@ runcmd:
     bash /root/prewarm-cache.sh || true
   - touch /root/.hushlogin
   - |
-    for dir in .act .aspnet .azure .bash_aliases .bashrc .cache .config .claude .claude.json .digrc .dotnet .gitconfig .hushlogin .kube .lacework.toml .launchpadlib .local .npm .npmrc .nuget .oh-my-posh .oh-my-zsh .opencommit .p10k.zsh .profile .terraform.d .tfenv .tmux .tmux.conf .vim .vimrc .vscode .vscode-insiders .vscode-server .vscode-server-insiders .wget-hsts .z .zshrc dotfiles 40docs go bin snap; do
+    # Copy user configuration to /etc/skel for new users
+    # Exclude .oh-my-zsh to prevent conflicts with dotfiles installation (issue #319)
+    for dir in .act .aspnet .azure .bash_aliases .bashrc .cache .config .claude .claude.json .digrc .dotnet .gitconfig .hushlogin .kube .lacework.toml .launchpadlib .local .npm .npmrc .nuget .oh-my-posh .opencommit .p10k.zsh .profile .terraform.d .tfenv .tmux .tmux.conf .vim .vimrc .vscode .vscode-insiders .vscode-server .vscode-server-insiders .wget-hsts .z .zshrc dotfiles 40docs go bin snap; do
       [ -e /root/$dir ] && cp -a /root/$dir /etc/skel 2>/dev/null || true
     done
+    # Note: .oh-my-zsh is intentionally excluded from the list above to allow 
+    # dotfiles installation to handle Oh My Zsh setup without conflicts
     mkdir -p /root/.claude
     if [ -f /root/.claude/mcp.json ]
     then

